cg/cli/workflow/balsamic/base.py

```diff
@@ -149,7 +136,7 @@ def config_case(
             # figure out flowcell name from header
             with context.obj["gzipper"].open(file_obj.full_path) as handle:
                 header_line = handle.readline().decode()
-                header_info = context.obj["analysis_api"].fastq_header(header_line)
+                header_info = context.obj["fastq_api"].parse_header(header_line)
```

Comment:
This code block is kind of complex and very hard to test. I suggest to split this up and move to functions